frontend/src/component/integrations/IntegrationForm/IntegrationParameters/IntegrationParameter/IntegrationParameterEnableWithDropdown.tsx

```diff
@@ -0,0 +1,78 @@
+import { StyledAddonParameterContainer } from '../../IntegrationForm.styles';
+import GeneralSelect from 'component/common/GeneralSelect/GeneralSelect';
+import { AddonParameterSchema, AddonSchema } from 'openapi';
+import { ChangeEventHandler } from 'react';
+import { IntegrationParameterTextField } from './IntegrationParameterTextField';
+import { useState } from 'react';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+
+export interface IIntegrationParameterEnableWithDropdownProps {
+    parametersErrors: Record<string, string>;
+    definition: AddonParameterSchema;
+    setParameterValue: (param: string) => ChangeEventHandler<HTMLInputElement>;
+    config: AddonSchema;
+    enableTitle: string;
+    enableOptionText: string;
+    disableOptionText: string;
+}
+
+export const IntegrationParameterEnableWithDropdown = ({
+    definition,
+    config,
+    parametersErrors,
+    setParameterValue,
+    enableTitle,
+    enableOptionText,
+    disableOptionText,
+}: IIntegrationParameterEnableWithDropdownProps) => {
+    const value = config.parameters[definition?.name] || '';
+    const renderProjectOptions = () => [
+        { key: 'disabled', label: disableOptionText },
+        { key: 'enabled', label: enableOptionText },
+    ];
```

Comment:
Any reason why this is a function instead of simply an array?
 
```ts
const renderProjectOptions = [
  { key: 'disabled', label: disableOptionText },
  { key: 'enabled', label: enableOptionText },
];
```

Reply:
that's a good point!